jaxrs-api/src/test/java/javax/ws/rs/client/ClientBuilderStub.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2018 IBM and Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package javax.ws.rs.client;
+
+import java.security.KeyStore;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.Configuration;
+import javax.ws.rs.core.Feature;
+
+/**
+ * {@link javax.ws.rs.client.ClientBuilderListener} unit tests.
+ *
+ * @author Andy McCright
+ * @since 2.2
+ */
+public class ClientBuilderStub extends ClientBuilder {
+
+    ConfigurationStub config = new ConfigurationStub();
+    /* (non-Javadoc)
+    * @see javax.ws.rs.core.Configurable#getConfiguration()
+    */
+    @Override
+    public Configuration getConfiguration() {
+    return config;
+    }
+
+    /* (non-Javadoc)
+     * @see javax.ws.rs.core.Configurable#property(java.lang.String, java.lang.Object)
+     */
+    @Override
+    public ClientBuilder property(String key, Object value) {
+        config.properties.put(key, value);
+        return this;
+    }
+
+    /* (non-Javadoc)
+     * @see javax.ws.rs.core.Configurable#register(java.lang.Class)
+     */
+    @Override
+    public ClientBuilder register(Class<?> clazz) {
+        config.providerClasses.put(clazz, 5000);
+        return this;
+    }
+
+    /* (non-Javadoc)
+     * @see javax.ws.rs.core.Configurable#register(java.lang.Object)
+     */
+    @Override
+    public ClientBuilder register(Object o) {
+        if (o instanceof Feature) {
+            config.features.add((Feature)o);
+        } else {
+            config.providerInstances.put(o, 5000);
+        }
+        return this;
+    }
+
+    /* (non-Javadoc)
+     * @see javax.ws.rs.core.Configurable#register(java.lang.Class, int)
+     */
+    @Override
+    public ClientBuilder register(Class<?> arg0, int arg1) {
+        // TODO Auto-generated method stub
```

Comment:
I assume you will overhaul before the actual merge to get rid of all those TODOs?

Reply:
That wasn't my plan.  I would prefer to leave the TODOs there so it is clear for others that these methods don't do anything.  If somebody else were to re-use the `ClientBuilderStub`'s unimplemented methods for other test cases, then they would need to implement them.

Reply:
Can we at least remove the word "TODO"? I really looks like you simply forgot to do an open task.

Reply:
Sure, I replaced it with `// unimplemented`.